quinn-runtime/src/lib.rs

```diff
@@ -0,0 +1,71 @@
+#[cfg(feature = "runtime-tokio")]
+mod tokio_runtime;
+#[cfg(feature = "runtime-tokio")]
+pub use tokio_runtime::*;
+
+#[cfg(feature = "runtime-async-std")]
+mod async_std_runtime;
+#[cfg(feature = "runtime-async-std")]
+pub use async_std_runtime::*;
+
+use std::fmt::Debug;
+use std::future::Future;
+use std::io;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+use std::time::Instant;
+
+pub trait AsyncWrappedUdpSocket: Sized + Send + Debug {
```

Comment:
Why the cfg gates on these methods?

Reply:
As the comments explain, on Unix we expect to be able to do more specific operations in order to support ECN. On non-unix the tokio backend can not provide these methods as there is no AsyncFd on non-unix, and the tokio::net::UdpSocket does not provide a way to get a reference to the underlying std::net::UdpSocket (if there is even one?).

Reply:
> there is no AsyncFd on non-unix

Perhaps something could be implemented in terms of e.g. https://docs.rs/tokio/latest/tokio/net/struct.UdpSocket.html#method.ready instead?

> the tokio::net::UdpSocket does not provide a way to get a reference to the underlying std::net::UdpSocket

What if we exposed the underlying file descriptor directly instead?

Reply:
AsRawFd also only exists on Unix. I don't think we can get around this problem easily. The interface we are describing here is completely different for Unix and Non-Unix platforms.